### DISCUSSION TITLE
Yuzu, Nintendo Switch Pro Controller, Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by bt0070069** [v1993/evdevhook2#11] [Q&A]
Hi,

I followed the steps in the instructions, and it doesn't seem to work in Yuzu (using Ubuntu 22.04). The Nintendo Switch Pro Controller paired out of the box without the evdevhook2, but without motion control support in yuzu. After installing the evdevhook2 nothing has changed. Any help would be appreciated.

Thanks.

**Comment by v1993** [comment]
What does evdevhook2 output when you run it? It will log all devices it
exposes. I'd expect this to be an access rights issue (which I'm well aware
should be better communicated to user, see #10).

Worth noting is that yuzu is capable of using motion controls of Switch
controllers without the need for external motion server (and with somewhat
better latency), provided you configure access correctly with udev rules.

**Comment by bt0070069** [comment]
Hi,

When I open the terminal and type in evdevhook2 nothing happens, it just
hangs and nothing is displayed. I have to CTRL - C to get back to the
prompt.

In issue #9 the solution was to use sudo evdevhook2, which works, and I was
able to get yuzu to connect to the UDP server, and detect the motion in my
Nintendo Switch Pro Controller.

Thanks for your assistance!

I could not get yuzu to detect motion without evdevhook2. The only option
was to use a UDP server. I'm curious how to get it to detect motion without
it, like you said it is supposed to.

**Comment by v1993** [comment]
Like I said - you need to create a udev rule to allow non-root access to
hidraw nodes. There's a pinned message in linux channel of yuzu discord by
me with one such rule.

As a side note - I would not recommend running evdevhook2 as root, but
rather fixing access rights with udev files for driver-created nodes as
well if you want to use evdevhook2.

**Comment by bt0070069** [comment]
I am not sure how to fix access rights with udev files for driver-created
nodes.

Also, I went on the yuzu website (https://yuzu-emu.org/), and got a link to
their discord, however it did not contain a Linux channel. The only
channels on the yuzu discord I could see were announcements and read-me.

**Reply by goldenx86** [reply]
Read the read-me channel carefully.

**Comment by v1993** [comment]
There is some sort of confirmation required to fully join. It's been an
eternity since I've joined, so you'll have to figure this out yourself.

**Comment by bt0070069** [comment]
I'll just run evdevhook2 with root access. I'm not concerned about it.

**Comment by bt0070069** [comment]
Thank you
